target/linux/ramips/dts/mt7621_beeline_smartbox-turbo-plus.dts

```diff
@@ -0,0 +1,213 @@
+		};
+
+		led_status_green: status_green {
+			gpios = <&gpio 15 GPIO_ACTIVE_HIGH>;
+			label = "smartbox-turbo-plus:green:status";
+		};
+
+		led_status_red: status_red {
+			gpios = <&gpio 16 GPIO_ACTIVE_HIGH>;
+			label = "smartbox-turbo-plus:red:status";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&nand {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "Bootloader";
+			reg = <0x0 0x80000>;
+			read-only;
+		};
+
+		partition@80000 {
+			label = "Config";
+			reg = <0x80000 0x40000>;
+		};
+
+		partition@c0000 {
+			label = "Bdata";
+			reg = <0xc0000 0x40000>;
+			read-only;
+		};
+
+		factory: partition@100000 {
+			label = "factory";
+			reg = <0x100000 0x40000>;
+			read-only;
+		};
+
+		partition@140000 {
+			label = "crash";
+			reg = <0x140000 0x40000>;
+			read-only;
+		};
+
+		partition@180000 {
+			label = "crash_syslog";
+			reg = <0x180000 0x40000>;
+			read-only;
+		};
+
+		partition@1c0000 {
+			label = "reserved0";
+			reg = <0x1c0000 0x40000>;
+			read-only;
+		};
+
+		/* uboot expects to find kernels at 0x200000 & 0x600000
+		 * referred to as system 1 & system 2 respectively.
+		 * a kernel is considered suitable for handing control over
+		 * if its linux magic number exists & uImage CRC are correct.
+		 * If either of those conditions fail, a matching sys'n'_fail flag
+		 * is set in uboot env & a restart performed in the hope that the
+		 * alternate kernel is okay.
+		 * if neither kernel checksums ok and both are marked failed, system 2
+		 * is booted anyway.
+		 *
+		 * Note uboot's tftp flash install writes the transferred
+		 * image to both kernel partitions.
+		 */
+
+		partition@200000 {
+			label = "kernel_stock";
+			reg = <0x200000 0x400000>;
+		};
+
+		partition@600000 {
+			label = "kernel";
+			reg = <0x600000 0x400000>;
+		};
+
+		/* ubi partition is the result of squashing
+		 * next consecutive stock partitions:
+		 * - rootfs0 (rootfs partition for stock kernel0),
+		 * - rootfs1 (rootfs partition for stock failsafe kernel1),
+		 * - overlay (used as ubi overlay in stock fw)
+		 * resulting 117,5MiB space for packages.
+		 */
+
+		partition@a00000 {
+			label = "ubi";
+			reg = <0xa00000 0x7580000>;
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "pci14c3,7615";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
+
+&pcie1 {
+	wifi@0,0 {
+		compatible = "pci14c3,7603";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+		ieee80211-freq-limit = <2400000 2500000>;
+	};
+};
+
+&gmac0 {
+	mtd-mac-address = <&factory 0x4>;
```

Comment:
Nothing in 0x28, 0x2e, 0xe000, 0xe006 ...?

Reply:
Hi @MaxS0niX ,

I have an similar device from my ISP. Thansk for the work you are doing here. 

From what I saw on the 4pda forum the mac address should be in "&factory 0x21000". Do you get the correct IP address?

You should also increment the wirleess by 2 and 3 for example as this is what is done in my device stock firmware.

Reply:
> Hi @MaxS0niX ,
> 
> I have an similar device from my ISP. Thansk for the work you are doing here.
> 
> From what I saw on the 4pda forum the mac address should be in "&factory 0x21000". Do you get the correct IP address?
> 
> You should also increment the wirleess by 2 and 3 for example as this is what is done in my device stock firmware.

What do you have for the device? 
Device get the correct MAC for all interfaces.
MAC LAN = AA:BB:CC:DD:EE:F**2** or AA:BB:CC:DD:EE:F**C**
MAC WAN = MAC LAN+1
MAC Wi-Fi (2.4 GHz) = MAC LAN
MAC Wi-Fi (5 GHz) = MAC LAN+2

Reply:
Please don't resolve issues when they are not resolved. The result of this discussion is that we don't know which MAC address assignment is correct.

Closing unresolved issues will look like you try to hide shortcomings of your commit (and is very annoying for, because for people like this I have to reopen anything and check what else should be hidden from my view ...)

Reply:
> Please don't resolve issues when they are not resolved. The result of this discussion is that we don't know which MAC address assignment is correct.
> 
> Closing unresolved issues will look like you try to hide shortcomings of your commit (and is very annoying for me, because for people like this I have to reopen anything and check what else should be hidden from my view ...)

Sorry, I thought this was resolved in the past [PR](https://github.com/openwrt/openwrt/pull/3435#discussion_r491474100).

Reply:
> Nothing in 0x28, 0x2e, 0xe000, 0xe006 ...?

In "Factory" i looked at three devices:
```
0x4 		- LAN MAC
0x8004		- LAN MAC +2 (Wi-Fi 5 GHz)
0x21000		- LAN MAC
0x21010		- Serial number on the backplate stick
0x21050		- ?Version
0x21060		- PCBASN - Serial number on PCB stick
0x21080		- SSID 2G
0x210A0		- PASS 2G

No adresses
0xe000	- FF FF FF FF FF FF
0xe006 	- FF FF FF FF FF FF
0x28 	- FD 9B FF FF FF FF
0x2e 	- FF FF FF FF FF FF
```
If required, I can send an image of mtd2.

Reply:
Command output # sc_fl_map in U-boot:
```
Name                Offset      Length      
calibration1        200000      1000        
calibration2        208000      1000        
mac                 221000      6           
CSN                 221010      c           
language ID         221040      4           
Domain ID           221050      4           
PCBASN              221060      c           
PIN                 221070      8           
SSID                221080      20          
Passphrase          2210a0      40          
```